api/log.go

```diff
@@ -47,6 +47,17 @@ import (
 //   description: Build number
 //   required: true
 //   type: integer
+// - in: query
+//   name: page
+//   description: The page of results to retrieve
+//   type: integer
+//   default: 1
+// - in: query
+//   name: per_page
+//   description: How many results per page to return
+//   type: integer
+//   maximum: 500
+//   default: 100
```

Comment:
This doc comment is a bit clearer on my intention. @jbrockopp do you want the maximum to be 100? I would like to increase that for this endpoint.

Reply:
Yeah, we shouldn't allow anything more than 100 results per page 👍

Especially for something like logs which can be quite taxing to retrieve depending on the size

Reply:
even 100 seems like a lot, tbh.

Reply:
I dropped the maximum back to 100.

And I left the default at 100 so that it is backwards compatible. Does that look better?

Reply:
@cognifloyd apologies if my initial review left you under the impression that this breaking change is bad.

More so, I wanted to confirm it because if so, we try to update the PR title with a `!`

i.e. `feat(api/log): support paging on GetBuildLogs` -> `feat(api/log)!: support paging on GetBuildLogs`

FWIW - I'm not opposed to making the default `10` instead of current behavior of `100`.

Like @wass3rw3rk mentioned, `100` does seem like a lot since it can be taxing on the system.

i.e. someone has a build with a bunch of steps that each have > `1 MB` in logs

For now, I'll plan to approve this PR so if we don't want to change it, then we can consider it in the future.

Reply:
Ok. I can add `!` (I'm slowly learning this commit format) and switch to 10 by default. I'll do that in 30 min or so

Reply:
Done. I lowered the `per_page` default to 10 and marked this PR as a breaking change.

Reply:
thanks.. i like it. we can either hardcode in CLI as @jbrockopp said and/or consider dropping hints if they are viewing a subset of available data.